client/src/components/Payment/Payment.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import { useStripe, useElements, CardNumberElement, CardExpiryElement, CardCvcElement, } from '@stripe/react-stripe-js';
+import { addCardToCustomer } from '../../helpers/APICalls/stripe'
+import { useAuth } from '../../context/useAuthContext';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import useStyles from './useStyles'
+
+export default function Payment(): JSX.Element {
+    const stripe = useStripe();
+    const elements = useElements();
+    const classes = useStyles();
+    const { loggedInUser } = useAuth();
+    const { updateSnackBarMessage } = useSnackBar();
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        if (!stripe || !elements) {
+            return;
+        }
+
+        const cardElement = elements.getElement(CardNumberElement);
+
+        if (loggedInUser && cardElement) {
+            stripe
+                .createPaymentMethod({
+                    type: 'card',
+                    card: cardElement,
+                    billing_details: {
+                        name: loggedInUser.username,
+                    },
+                })
+                .then(function (result) {
+                    if(result.paymentMethod){
+                        const cardId = result.paymentMethod.id;
+                        const stripeId = loggedInUser.stripeId;
+                        addCardToCustomer(cardId, stripeId);
+                        updateSnackBarMessage("Card has been added to your account.");
+                    } else {
+                        updateSnackBarMessage("Card could not be added");
+                    }
+                })
+        }
+    };
+
+    return (
+        <>
+            <Grid container justifyContent="center">
+                <form className={classes.form} onSubmit={handleSubmit}>
+                    <Grid container className={classes.formGrid}>
+                        <Grid className={classes.paper}>
```

Comment:
I noticed that you are using a flexbox display for this Grid, rather than using `Grid` consider using `Box`, it has flexbox built into it.